src/test/fakeExtensionContext.ts

```diff
@@ -54,14 +56,32 @@ export class FakeExtensionContext implements ExtensionContext {
     public static async getNew(): Promise<FakeExtensionContext> {
         const ctx = new FakeExtensionContext()
         try {
-            ExtensionDisposableFiles.getInstance().dispose()
+            TestExtensionDisposableFiles.clearInstance()
         } catch {
-            await ExtensionDisposableFiles.initialize(ctx)
+            // Ignore.
         }
+        await TestExtensionDisposableFiles.initialize(ctx)
```

Comment:
TestExtensionDisposableFiles doesn't appear to have a static initialize method, which was confusing when I initially read this change. Consider changing to ExtensionDisposableFiles.initialize(...)

Reply:
but if TestExtensionDisposableFiles later adds initialize(), then the caller must be updated.

Reply:
Sounds like a problem to solve if/when that happens

Reply:
Doesn't that apply to all uses of subclasses?